setup.py

```diff
@@ -74,7 +74,8 @@ def get_version():
     license='BSD License',
     packages=['markdown', 'markdown.extensions'],
     python_requires='>=3.6',
-    install_requires=["importlib-metadata>=4.4;python_version<'3.10'"],
+    install_requires=["importlib-metadata >=3.6",
+                      "importlib-metadata >=4.4; python_version >= '3.10'"],
```

Comment:
After taking another look at this, I question the usefulness of this. Note this code:

https://github.com/Python-Markdown/markdown/blob/af38c42706f8dff93694d4a7572003dbd8b0ddc0/markdown/util.py#L32-L36

We only import `importlib-metadata` for `python_version < 3.10`. Therefore, we don't need `importlib-metadata` at all for `python_version >= 3.10`. Lets add `python_version < '3.10'` to the first line and remove the second line.

I now remember why we didn't do this earlier. We certainly cannot go back any further than 3.6 as we need to use `group`. As a reminder, in Python 3.10, `group` is the only supported method, so we have to use group to support Python 3.10. While support for `group` was first added in 3.6, my understanding is that it was in beta until 4.4, which is the first version which is guaranteed to have behavior which matches what is found in Python 3.10. Sure, maybe it all seems to work in our limited testing, but can we be confident that it will always work as expected in all cases for all users? I can only be reasonably sure if we require 4.4.

And finally, does 3.6 even give you what you want? I forget which version flake8 is pinned at.

Reply:
🤔 flake8 is limited to `importlib-metadata<4.3` 
So the >=3.6 constraints in a python<3.10 should be resolved by a 4.2 install with flake8 by dependencies resolver tools like pip/poetry.
I think that tox could solve the "can it work with all combinations of python version & importlib-metadata range library?" by empirical testing if we don't trust importlib-metadata documentation.

Reply:
> if we don't trust importlib-metadata documentation.

The issue is that Python 3.10's [Release Notes][1] states that Python 3.10 has feature parity with import-metadata 4.6. Therefore, my assumption is that any earlier version of importlib-metadata will not behave exactly as expected in every edge case. And I'm not interested in testing every edge case. Therefore, we will not be supporting any earlier version.

As an aside, I see that we pinned 4.4. It appears we should have pinned 4.6. If we change anything, it will only be to pin 4.6 for `Python < 3.10`.

[1]: https://docs.python.org/3/whatsnew/3.10.html#importlib-metadata